components/email.php

```diff
@@ -0,0 +1,186 @@
+<?php
+/**
+ * Manage BuddyPress Email Post Types.
+ *
+ * @since 1.6.0
+ */
+class BPCLI_Email extends BPCLI_Component {
+	/**
+	 * Create a new email post connected to an email type.
+	 *
+	 * ## OPTIONS
+	 *
+	 * --type=<type>
+	 * : Email type for the email (should be unique identifier, sanitized like a post slug).
+	 *
+	 * [--type-description=<type-description>]
+	 * : Email type description.
+	 *
+	 * --subject=<subject>
+	 * : Email subject line. Email tokens allowed. View https://codex.buddypress.org/emails/email-tokens/ for more info.
+	 *
+	 * --content=<content>
+	 * : Email content. Email tokens allowed. View https://codex.buddypress.org/emails/email-tokens/ for more info.
+	 *
+	 * [--plain-text-content=<plain-text-content>]
+	 * : Plain-text email content. Email tokens allowed. View https://codex.buddypress.org/emails/email-tokens/ for more info.
+	 *
+	 * ## EXAMPLES
+	 *
+	 *     $ wp bp email create --type=new-event --type-description="Send an email when a new event is created" --subject="[{{{site.name}}}] A new event was created" --content="<a href='{{{some.custom-token-url}}}'></a>A new event</a> was created" --plain-text-content="A new event was created"
+	 *     Success: Email post created for type "new-event".
+	 *
+	 * @alias add
+	 */
+	public function create( $args, $assoc_args ) {
+		$switched = false;
+
+		if ( false === bp_is_root_blog() ) {
+			$switched = true;
+			switch_to_blog( bp_get_root_blog_id() );
+		}
+
+		$term = term_exists( $assoc_args['type'], bp_get_email_tax_type() );
+
+		// Term already exists so don't do anything.
+		if ( $term !== 0 && $term !== null ) {
+			if ( true === $switched ) {
+				restore_current_blog();
+			}
+
+			WP_CLI::error( "Email type '{$assoc_args['type']}' already exists." );
+		}
+
+		if ( ! empty( $args[0] ) ) {
+			$assoc_args['content'] = $this->read_from_file_or_stdin( $args[0] );
+		}
+		if ( \WP_CLI\Utils\get_flag_value( $assoc_args, 'edit' ) ) {
+			$input = \WP_CLI\Utils\get_flag_value( $assoc_args, 'content', '' );
+			if ( $output = $this->_edit( $input, 'WP-CLI: New BP Email Content' ) ) {
+				$assoc_args['content'] = $output;
+			} else {
+				$assoc_args['content'] = $input;
+			}
+		}
+
+		$id = $assoc_args['type'];
+
+		$defaults = array(
+			'post_status' => 'publish',
+			'post_type'   => bp_get_email_post_type(),
+		);
+
+		$email = array(
```

Comment:
This all suggests to me that BP needs a single function for generating email posts. See `bp_core_install_emails()`. Not a blocker for this PR, but a future enhancement.

Reply:
Agreed.

Reply:
+1